tf2_ros/src/static_transform_broadcaster_program.cpp

```diff
@@ -40,87 +40,126 @@
-      msg.transform.translation.y = atof(argv[2]);
-      msg.transform.translation.z = atof(argv[3]);
+    msg.header.stamp = rclcpp::Time::now();
+    msg.header.frame_id = argv[7];
+    msg.child_frame_id = argv[8];
+  }
+  // else if (argc == 2) {
+  //   const std::string param_name = argv[1];
+  //   ROS_INFO_STREAM("Looking for TF in parameter: " << param_name);
+  //   XmlRpc::XmlRpcValue tf_data;
 
-      tf2::Quaternion quat;
-      quat.setRPY(atof(argv[6]), atof(argv[5]), atof(argv[4]));
-      msg.transform.rotation.x = quat.x();
-      msg.transform.rotation.y = quat.y();
-      msg.transform.rotation.z = quat.z();
-      msg.transform.rotation.w = quat.w();
+  //   if (!ros::param::has(param_name) || !ros::param::get(param_name, tf_data)) {
+  //     ROS_FATAL_STREAM("Could not read TF from parameter server: " << param_name);
+  //     return -1;
+  //   }
 
-      msg.header.stamp = rclcpp::Time::now();
-      msg.header.frame_id = argv[7];
-      msg.child_frame_id = argv[8];
-    }
-    rclcpp::WallRate loop_rate(0.2);
-    while (rclcpp::ok())
-    {
-      //TODO(tfoote) reimplement latching
-      ROS_INFO("LOOPING due to no latching at the moment\n");
-      broadcaster.sendTransform(msg);
-      rclcpp::spin_some(node);
-      loop_rate.sleep();  
-    }
+  //   // Check that all required members are present & of the right type.
+  //   if (!validateXmlRpcTf(tf_data)) {
+  //     ROS_FATAL_STREAM("Could not validate XmlRpcC for TF data: " << tf_data);
+  //     return -1;
+  //   }
 
-    // broadcaster.sendTransform(msg);
-    // ROS_INFO("Spinning until killed publishing %s to %s", msg.header.frame_id.c_str(), msg.child_frame_id.c_str());
-    // rclcpp::spin(node);
-    return 0;
-  }
+  //   msg.transform.translation.x = (double) tf_data["transform"]["translation"]["x"];
+  //   msg.transform.translation.y = (double) tf_data["transform"]["translation"]["y"];
+  //   msg.transform.translation.z = (double) tf_data["transform"]["translation"]["z"];
+  //   msg.transform.rotation.x = (double) tf_data["transform"]["rotation"]["x"];
+  //   msg.transform.rotation.y = (double) tf_data["transform"]["rotation"]["y"];
+  //   msg.transform.rotation.z = (double) tf_data["transform"]["rotation"]["z"];
+  //   msg.transform.rotation.w = (double) tf_data["transform"]["rotation"]["w"];
+  //   msg.header.stamp = ros::Time::now();
+  //   msg.header.frame_id = (std::string) tf_data["header"]["frame_id"];
+  //   msg.child_frame_id = (std::string) tf_data["child_frame_id"];
+  // }
   else
   {
     printf("A command line utility for manually sending a transform.\n");
     //printf("It will periodicaly republish the given transform. \n");
     printf("Usage: static_transform_publisher x y z qx qy qz qw frame_id child_frame_id \n");
     printf("OR \n");
     printf("Usage: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id \n");
+    printf("OR \n");
+    printf("Usage: static_transform_publisher /param_name \n");
```

Comment:
probably this should be commented too if we're commenting the rest of https://github.com/ros/geometry2/commit/b6d74e52086bdbb5dd62fd423c26f8caf16a01d6